framework/src/modules/chain/dpos/apply.js

```diff
@@ -0,0 +1,211 @@
+const { getAddressFromPublicKey } = require('@liskhq/lisk-cryptography');
+const BigNum = require('@liskhq/bignum');
+const {
+	updateProducedBlocks,
+	updateMissedBlocks,
+	distributeRewardsAndFees,
+} = require('./common');
+
+class Apply {
+	constructor({
+		storage,
+		slots,
+		activeDelegates,
+		logger,
+		delegates,
+		exceptions,
+	}) {
+		this.storage = storage;
+		this.slots = slots;
+		this.activeDelegates = activeDelegates;
+		this.logger = logger;
+		this.delegates = delegates;
+		this.exceptions = exceptions;
+	}
+
+	/**
+	 * @param {Block} block
+	 */
+	async apply(block, tx) {
+		const round = this.slots.calcRound(block.height);
+
+		await this.updateProducedBlocks(block, tx);
+
+		if (this.hasRoundFinished(block)) {
+			const summarizedRound = await this.summarizeRound(block, tx);
+			const nonForgedDelegateAddresses = await this.getNonForgedDelegateAddresses(
+				round,
+				summarizedRound.delegatePks
+			);
+			await this.updateMissedBlocks(nonForgedDelegateAddresses, tx);
+			const updatedDelegates = await this.distributeRewardsAndFees(
+				summarizedRound,
+				tx
+			);
+			await this.updateVotes(updatedDelegates, tx);
+		}
+	}
+
+	async updateProducedBlocks(block, tx) {
```

Comment:
if it's private function, it should be `_ updateProducedBlocks`
Same apply to belows